lib/test/apyfixedarray/test_methods.py

```diff
@@ -121,3 +121,134 @@ def test_round_trip_conversion():
     for bit_pattern in range(256):
         a = APyFixedArray([bit_pattern], int_bits=4, frac_bits=4)
         assert (APyFixedArray.from_float(a.to_numpy(), 4, 4)).is_identical(a)
+
+
+def test_reshape():
+    nums = [
+        5,
+        10,
+        15,
+        20,
+        25,
+        30,
+        35,
+        40,
+        45,
+        50,
+        55,
+        60,
+    ]
+
+    # Creating 1D array to be reshaped
+    arr = APyFixedArray(nums, bits=20, int_bits=10)
+
+    possible_shapes = [(1, 12), (2, 6), (3, 4), (4, 3), (6, 2), (12, 1)]
+    possible_shapes += [(-1, 12), (2, -1), (3, -1), (4, -1), (-1, 3), (-1, 2), (12, -1)]
+
+    for shape in possible_shapes:
+        reshaped_arr = arr.reshape(shape)
+        for i, row in enumerate(reshaped_arr):
+            for j, float_ in enumerate(row):
+                arr_index = i * reshaped_arr.shape[1] + j
+                if not APyFixed.is_identical(arr[arr_index], float_):
+                    pytest.fail(f"Mismatch at index {arr_index} during reshape")
+
+        go_back = reshaped_arr.reshape(arr.shape)
+        if not APyFixedArray.is_identical(go_back, arr):
+            pytest.fail(f"Mismatch after reshaping back at index {arr_index}")
+
+    possible_3d_shapes = [
```

Comment:
If nothing else, I would split this test into the different cases, so `test_reshape_2d`, `test_reshape_3d`, ...